### ISSUE TITLE
From Arduino IDE, FSPage does not show any page

### ISSUE THREAD

**Issue by nostradamux** [Hieromon/PageBuilder#35]
Hello.
I have tested in Arduino IDE PageBuilder for d1_mini, and only FSPAge is not working. 
It is the one that data is needed, so my question is: any hint to solve the issue? ( I have tested sent data to fileSystem)
Thanks

**Comment by Hieromon** [comment]
@nostradamux Hello.
I reproduced the issue. A breaking change in ESP8266 3.0.0 has degraded the [PageStream](https://github.com/Hieromon/PageBuilder/blob/master/src/PageStream.h) class. I missed this change.
So, I merged the fixed version into the master. Can you test it in your environment?, If the result is fine, I will release a fix as PageBuilder v1.5.1.

Thank you for your contribution.

**Comment by nostradamux** [comment]
Hello @Hieromon
Thanks for your quick response and update.
I have change code to 1.5.1 present in master, but I having the same behavior. Activating PD_DEBUG, I can see that it is not possible to open files from data, but I would say they are in.

esp-ap started. IP:192.168.4.1
[event] SoftAP station connected.
[PB] HTTP_GET /
[PB] Buf preserve:16, Free heap:39464
[PB] _mold /root.htm open failed


I see the same problem using Arduino "Esp8266 Sketch Data Upload" utility and using Upload SPIFFS image in platformIo.


I would suspect that formatting is not correct (sorry I am confused with the use of LittleFS, SPIFFS, FS, etc...). Could be this the issue? How could confirm that data is correctly flashed into board with proper FS (at least the ones needed for PageBuilder?

Thanks

**Comment by Hieromon** [comment]
@nostradamux 
[SPIFFS, the ESP8266 file system, has been deprecated](https://arduino-esp8266.readthedocs.io/en/latest/filesystem.html#spiffs-deprecation-warning) since ESP8266 Arduino core v2.7.1. That's why **PageBuilder uses LittleFS as its default file system**. Also, the uploader for the files contained in the data folder to the LittleFS file system is different from SPIFFS.

If you continue to use SPIFFS, you need to enable the *PB_USE_SPIFFS* macro definition.
Refs.: https://github.com/Hieromon/PageBuilder#significant-changes
On the other hand, if you want to migrate your Arduino IDE environment to LittleFS, please follow the guide [here](https://github.com/earlephilhower/arduino-esp8266littlefs-plugin#installation) to install the [Arduino ESP8266 LittleFS Filesystem Uploader](https://github.com/earlephilhower/arduino-esp8266littlefs-plugin). To upload files to LittleFS use `Tools > ESP8266 LittleFS Data Upload`.

However, PlatformIO takes the opposite approach. That seems to be confusing to you. **In PlatformIO, the default file system is still SPIFFS**. They say that "The SPIFFS file system is used by default in order to keep legacy project compatible".
Refs.: https://docs.platformio.org/en/latest/platforms/espressif8266.html#using-filesystem
In order to use LittleFS with PlatformIO, you need to write an explicit declaration in `platformio.ini` file.

```ini
[env:d1_mini]
platform = espressif8266
framework = arduino
board = ...
board_build.filesystem = littlefs
```

And, use `Upload File System image` task in PlatformIO IDE or `pio run --target` CLI command with uploadfs target.

**Comment by nostradamux** [comment]
Hello.
As quick measure, I have just change Arduino tools to upload efectively data in LittleFS format required by default by PageBuilder (sorry to not take it in account before).

Eventyally, FSPage detects correctly file names and also generates html/page, but content is weird:

!OTP tl hm> S86 iiJia{ wdh 0p; dsly niebok oiin eaie adn:02e .e; mri-ih:10m etain etr etdcrto:nn; clr FF akrudclr 067; bcgon-mg:-ektlna-rdettp 08b,#078; bcgon-mg:lna-rdett otm 08b,#078; bre-otm oi p 046; bre-ais p; bxsao:ist01x0rb(5,5,5,.) p p ga0 ,0 .9; } .sdls :cie{ bre-otm oi p 067; bxsao:001xrb(,0 ,02; } .m_ok{ dsly niebok it:3p; hih:2x agnlf:05m etclain ide akrud r(aaiaepgbs6,VOwKgAANUEgACAAgAAAzn0AAXIXAATALwApYACEEVYeVUURH2maNiKKCQc8JCCMFumVC9HCcFQWFg9WKiKssYoRFwNLpmx8CXtumxeed8vXvzLpOO5ekgiF0EUqSwC52xzGD0h3yLXRV6fZlGCQuk6/fhP04rW3LKqwHgo+PD1RF4oWBm3FwcAzi141cOEFoX2jePkhN6uwF5BoyegJbjZBkQtZqwr0WvJ09qQNZwuE/WNRA+uRqg4so13tDiwMlDStKfrAdxZd2zAfwkq4WEo7YFse8BNNZrWUsMtdAAqsaulS88BUB84QZ5qpApimFUA7qhY1doYEeNgacqAc1BPNBoYubtYaUEr4t7W8nTPDbKjlmlfbJfcmrP75s2WL1LxwyQsJZw+QQQXHTua9Bu1SnEdDN7XJ0d61nHHDFzMwgb9d6Gs//GbrAI1u419nZSzexAAEFkumC orpa; }<2Ji oWF P/2 dvcas"sdls"< rf"ji?sdMFba42&s_ye4>iir-66/>4/i>dvcas"sdls"< rf"ji?sdDRC-6H fiee 70pktp="DRC-6H fiee 70/>6/i>dvcas"sdls"< rf"ji?sdNia&s_ye8>an2/>0/i>dvcas"sdls"< rf"ji?sdMWF_GXj&s_ye4>III2_Qf/>8/i>dvcas"sdls"< rf"ji?sdMWF_GXj&s_ye4>III5_Qf/>2/i>dvcas"sdls"< rf"ji?sdMFba83&s_ye4>iir-04/>0/i>dvcas"sdls"< rf"ji?sdMFbaAF&s_ye4>iir-AC/>6/i>dvcas"sdls"< rf"ji?sdMVSA_17pktp="MVSA_17/>6/i>dvcas"sdls"< rf"ji?sdnsr_ii5&s_ye4>otaWf_G/>0/i>dvcas"sdls"< rf"ji?sdMVSA_80pktp="MVSA_80/>8/i>dvcas"sdls"< rf"ji?sdEORC&s_ye8>V2AE/>4/i>dvcas"sdls"< rf"ji?sdMFba18_X&s_ye8>iir-E3ET/>8/i>dvcas"sdls"< rf"ji?sdMVSA_E2pktp="MVSA_E2/>4/i>dvcas"sdls"< rf"ji?sdvdfnC5&s_ye8>oaoe38/>8/i>dvcas"sdls"< rf"ji?sdvdfnA42&s_ye4>oaoeAUC/>6/i>dvcas"sdls"< rf"ji?sdRCTL_07pktp="RCTL_07/>6/i><>ahe=/ ls=ln_utn>ecn/>/>я

(this is what I observe in explorer once device is connected to esp-ap and 192.168.4.1)

And here the log from console:

esp-ap started. IP:192.168.4.1
[event] SoftAP station connected.
[PB] HTTP_GET /
[PB] Buf preserve:16, Free heap:39728
[PB] _mold root.htm opened, SSID_LIST URI_ROOT
[PB] blk:4617
[PB] HTTP_GET /
[PB] Buf preserve:16, Free heap:37856
[PB] _mold root.htm opened, SSID_LIST URI_ROOT
[PB] blk:4945

Many thanks

**Comment by nostradamux** [comment]
Hello, again.

Using the solution proposed for PlatformIO gets correct results: I finally see the pages!. 
I will not investigate much more in Arduino IDE, as normally I work with PlatformIO. It is enough for me to have working PlatformIO.

Many thanks for your support!

**Comment by Hieromon** [comment]
@nostradamux Maybe PageBuilder update is failing.
I have released v1.5.1 and after a while, you will be able to check for updates from your Arduino IDE's Library Manager. Then try updating PageBuilder to v1.5.1. Maybe that should restore your IDE's library environment.

Anyway, the problem is solved, so I close it.
